### PR TITLE
Sql support to crud

### PR DESCRIPTION
**Pull request by YourTechBud**
I'm a bit skeptical about the sql update. Have a look at that.

### CODE REVIEW

File: crud/sql/update.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	"github.com/spaceuptech/space-cloud/utils"
+)
+
+// Update updates the document(s) which match the condition provided.
+func (s *SQL) Update(ctx context.Context, project, col string, req *model.UpdateRequest) error {
+	// Generate a prepared query builder
+	query := goqu.From(col).Prepared(true)
+	query = query.SetAdapter(goqu.NewAdapter(s.dbType, query))
+
+	if req.Find == nil {
```
Comment (Jayesh333) [inline crud/sql/update.go]:
```suggestion
	if req.Find != nil {
```

File: crud/sql/sql.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+	}
+
+	s.client = sql
+	return s, nil
+}
+
+// GetDBType returns the dbType of the crud block
+func (s *SQL) GetDBType() utils.DBType {
+	switch s.dbType {
+	case "postgres":
```
Comment (Jayesh333) [inline crud/sql/sql.go]:
```suggestion
	case "sql-postgres":
```

File: crud/sql/sql.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+		return utils.Postgres
+	case "mysql":
```
Comment (Jayesh333) [inline crud/sql/sql.go]:
```suggestion
	case "sql-mysql":
```

File: crud/sql/sql.go [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+		return utils.MySQL
+	}
+
+	return utils.MySQL
```
Comment (Jayesh333) [inline crud/sql/sql.go]: Shouldn't this return an error if no matching db found

File: crud/sql/sql.go [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+	s := &SQL{}
+
+	switch dbType {
+	case utils.Postgres:
+		sql, err = sqlx.Open("postgres", connection)
+		s.dbType = "postgres"
+
+	case utils.MySQL:
+		sql, err = sqlx.Open("mysql", connection)
+		s.dbType = "mysql"
```
Comment (Jayesh333) [inline crud/sql/sql.go]:
```suggestion
		s.dbType = "sql-mysql"
```

File: crud/sql/sql.go [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+func Init(dbType utils.DBType, connection string) (*SQL, error) {
+	var sql *sqlx.DB
+	var err error
+
+	s := &SQL{}
+
+	switch dbType {
+	case utils.Postgres:
+		sql, err = sqlx.Open("postgres", connection)
+		s.dbType = "postgres"
```
Comment (Jayesh333) [inline crud/sql/sql.go]:
```suggestion
		s.dbType = "sql-postgres"
```

File: crud/sql/sql.go (hunk 2 above)
Comment (YourTechBud) [reply crud/sql/sql.go]: its supposed to be without the "sql-" prefix. SQL.dbType is used directly in goqu

File: crud/sql/sql.go (hunk 3 above)
Comment (YourTechBud) [reply crud/sql/sql.go]: its supposed to be without the "sql-" prefix. SQL.dbType is used directly in goqu

File: crud/sql/sql.go (hunk 4 above)
Comment (YourTechBud) [reply crud/sql/sql.go]: It isn't possible that SQL.dbType would have any value apart from these two.

**Pull request merged**